interfaces/ldconfig/spec.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ldconfig
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Specification assists in collecting library directories associated with an
+// interface.
+//
+// Unlike the Backend itself (which is stateless and non-persistent) this type
+// holds internal state that is used by the ldconfig backend during the
+// interface setup process.
+type Specification struct {
+	// libDirs is the list of directories with libraries coming from
+	// different slots. The key has the form "<snap_name>.<slot>".
+	libDirs map[string][]string
+}
+
+// Methods called by interfaces
+
+// AddLibDirs add dirs with libraries to the specification.
+func (spec *Specification) AddLibDirs(snapName, slotName string, dirs []string) {
```

Comment:
we can capture the contextual info on the spec in AddConnectedPlug, I would expect that's the place we want this to work from? then we should error or panic if it's called with the contextual info unse

Reply:
see something like setScope in the apparmor backend, though what we need here is simpler

Reply:
Done now